resources/migrations/000_migrations.yaml

```diff
@@ -13933,6 +13933,207 @@ databaseChangeLog:
+              JOIN (
+              SELECT
+                id, JSON_OBJECTAGG(k, v) AS column_settings
+              FROM (
+              SELECT
+                id,
+                CASE
+                  WHEN JSON_EXTRACT(k, '$[0]') = 'ref'
+                  THEN
+                    CASE
+                      WHEN JSON_EXTRACT(k, '$[1][0]') = 'field-id'
+                      THEN CONCAT('["ref",["field",', JSON_EXTRACT(k, '$[1][1]'), ',null]]')
+                      WHEN JSON_EXTRACT(k, '$[1][0]') = 'field-literal'
+                      THEN CONCAT('["ref",["field","', JSON_EXTRACT(k, '$[1][1]'), '",{"base-type":"', JSON_EXTRACT(k, '$[1][2]'), '"}]]')
+                      WHEN JSON_EXTRACT(k, '$[1][0]') = 'fk->'
+                      THEN CONCAT('["ref",["field",',
+                                  CASE
+                                    WHEN SUBSTRING(JSON_EXTRACT(k, '$[1][2]'), 1, 1) = '['
+                                    THEN JSON_EXTRACT(k, '$[1][2][0]')
+                                    ELSE JSON_EXTRACT(k, '$[1][2]')
+                                  END,
+                                  ',{"source-field":',
+                                  CASE
+                                    WHEN SUBSTRING(JSON_EXTRACT(k, '$[1][1]'), 1, 1) = '['
+                                    THEN JSON_EXTRACT(k, '$[1][1][0]')
+                                    ELSE JSON_EXTRACT(k, '$[1][1]')
+                                  END,
+                                  '}]]')
+                      ELSE k
+                    END
+                  ELSE k
+                END AS k,
+                v
+              FROM (
+                SELECT
+                  id
+                  , JSON_UNQUOTE(JSON_EXTRACT(JSON_KEYS(settings), CONCAT('$[', idx, ']'))) AS k
+                  , JSON_EXTRACT(JSON_EXTRACT(settings, '$.*'), CONCAT('$[', idx, ']')) AS v
+                FROM (
+                    SELECT id, JSON_EXTRACT(visualization_settings, '$.column_settings') AS settings
+                    FROM report_card
+                    WHERE visualization_settings IS NOT NULL
+                      AND JSON_EXTRACT(visualization_settings, '$.column_settings') IS NOT NULL
+                ) AS column_settings
+                JOIN (
+                    SELECT a.x + b.x * 4 + c.x * 16 + d.x * 64 + e.x * 256 AS idx
+                    FROM (SELECT 0 AS x UNION SELECT 1 AS x UNION SELECT 2 AS x UNION SELECT 3 AS x) a
+                    JOIN (SELECT 0 AS x UNION SELECT 1 AS x UNION SELECT 2 AS x UNION SELECT 3 AS x) b
+                    JOIN (SELECT 0 AS x UNION SELECT 1 AS x UNION SELECT 2 AS x UNION SELECT 3 AS x) c
+                    JOIN (SELECT 0 AS x UNION SELECT 1 AS x UNION SELECT 2 AS x UNION SELECT 3 AS x) d
+                    JOIN (SELECT 0 AS x UNION SELECT 1 AS x UNION SELECT 2 AS x UNION SELECT 3 AS x) e
+                  ) AS indices -- 0 to 1023. Assumes there are no more than 1024 columns for every card.
```

Comment:
I'm making an assumption here that cards can't have more than 1024 columns. Is that reasonable? Do we define a limit on the number of columns anywhere?